tools/hive-bigquery/bigquery_component.py

```diff
@@ -0,0 +1,725 @@
+
+        dataset_ref = self.client.dataset(bq_table_model.dataset_id)
+        # Configures the load job
+        # job_config is of type: google.cloud.bigquery.job.LoadJobConfig
+        job_config = bigquery.LoadJobConfig()
+
+        if bq_table_model.is_partitioned:
+            # Specifies time-based partitioning for the table.
+            job_config.time_partitioning = bigquery.table.TimePartitioning(
+                type_=bigquery.table.TimePartitioningType.DAY,
+                field=bq_table_model.partition_column)
+            if bq_table_model.is_clustered:
+                # Fields defining clustering for the table
+                job_config.clustering_fields = bq_table_model.clustering_columns
+
+        # Configures the file format of the data
+        if bq_table_model.data_format == "ORC":
+            job_config.source_format = bigquery.SourceFormat.ORC
+        elif bq_table_model.data_format == "Parquet":
+            job_config.source_format = bigquery.SourceFormat.PARQUET
+        else:
+            job_config.source_format = bigquery.SourceFormat.AVRO
+
+        # Creates load job
+        self.client.load_table_from_uri(source_uri, dataset_ref.table(
+            bq_table_model.table_name), job_config=job_config, job_id=job_id)
+
+    def get_bq_table_row_count(self, bq_table_model, clause=''):
+        """Queries the migrated BigQuery table to get a count of rows
+
+        Args:
+            bq_table_model (:class:`BigQueryTableModel`): Wrapper to BigQuery
+                table details
+            clause (str): where clause to filter the table on partitions, if any
+
+        Returns:
+            int: Number of rows as an output from the query
+        """
+
+        query = "SELECT COUNT(*) AS n_rows FROM {0}.{1} {2}".format(
+            bq_table_model.dataset_id, bq_table_model.table_name, clause)
+        query_job = self.client.query(query)
+        results = query_job.result()
+        for row in results:
+            n_rows = row.n_rows
+            return n_rows
+
+    def load_gcs_to_bq(self, mysql_component, hive_table_model, bq_table_model):
+        """Loads data from GCS to BigQuery
+
+        Queries the tracking table and fetches information about the files
+        that have been copied to GCS and are ready to be loaded into
+        BigQuery, creates loading jobs and updates the job ID & job status in
+        the tracking table
+
+        Args:
+            mysql_component (:class:`MySQLComponent`): Instance of
+                MySQLComponent to connect to MySQL
+            hive_table_model (:class:`HiveTableModel`): Wrapper to Hive table
+                details
+            bq_table_model (:class:`BigQueryTableModel`): Wrapper to BigQuery
+                table details
+        """
+
+        print_and_log(
+            "Fetching information about files to load to BigQuery from "
+            "tracking table...")
+        query = "SELECT gcs_file_path FROM %s WHERE gcs_copy_status='DONE' " \
```

Comment:
Inconsistent use of String formatting. Let's switch this to new format if we can